lib/file-source/src/checkpointer.rs

```diff
@@ -154,13 +157,32 @@ impl CheckpointsView {
         }
     }
 
-    fn maybe_upgrade(&self, fresh: impl Iterator<Item = FileFingerprint>) {
-        for fng in fresh {
-            if let Some((_, pos)) = self
-                .checkpoints
-                .remove(&FileFingerprint::Unknown(fng.as_legacy()))
+    fn maybe_upgrade(
+        &self,
+        path: &Path,
+        fng: FileFingerprint,
+        fingerprinter: &Fingerprinter,
+        fingerprint_buffer: &mut Vec<u8>,
```

Comment:
This buffer does not appear to be used by any of the callers after calling the function. Could it be created within this function to avoid the additional parameter?

In fact, it doesn't look like the contents are used within this function either. I don't think this function is on the hot path (startup only) so reusing the buffer shouldn't be critical to performance. Am I missing something?

Reply:
Agreed, I was following the pattern of the other functions, but think we can just create the buffer in here since it is only called once at start-up.

Reply:
Actually, I realized the default buffer size 1 MB so I may replace this. I can see why we'd want to reuse them in-case there are a lot of files.